### ISSUE TITLE
Absorbtion is not invariant to order of args.

### ISSUE THREAD

**Issue by tomas789** [bastikr/boolean.py#112]
The function [absorb](https://github.com/bastikr/boolean.py/blob/431e5226c269e4c863cc851b971ca5e89fda04d7/boolean/boolean.py#L1446) should produce the same results no matter the order of the `args` passed to it.

There is a counter-example where this property does not hold.

```python
import boolean
        
algebra = boolean.BooleanAlgebra()
TRUE, FALSE, NOT, AND, OR, symbol = algebra.definition()

e = AND(OR(NOT(symbol("x1")), NOT(symbol("x2"))), TRUE, symbol("x2"), NOT(symbol("x1")))
args = [
    OR(NOT(symbol("x1")), NOT(symbol("x2"))),
    NOT(symbol("x1")),
    symbol("x2"),
]
print("ORIGINAL ORDER")
print(e.absorb(args))

args[1], args[2] = args[2], args[1]
print("SWAPPED")
print(e.absorb(args))
```

Which produces following output

```
ORIGINAL ORDER
[NOT(Symbol('x1')), Symbol('x2')]
SWAPPED
[NOT(Symbol('x1')), Symbol('x2'), NOT(Symbol('x1'))]
```

**Comment by tomas789** [comment]
After more investigation it looks like this is the core cause of the previous issue https://github.com/bastikr/boolean.py/issues/111

**Comment by pombredanne** [comment]
Fixed with #113 Thank you ++
